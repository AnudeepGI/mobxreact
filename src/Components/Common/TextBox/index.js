import React from 'react';

const TextBox = (props) => {

    return (<input className={props.className ? props.className : undefined}
        value={props.value ? props.value : undefined}
        onKeyPress={props.onKeyPress ? props.onKeyPress : undefined}
        onChange={props.onChange ? props.onChange : undefined} />)

}

export default TextBox;