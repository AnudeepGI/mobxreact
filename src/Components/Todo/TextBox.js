import React from 'react';

const TextBox = (props) => {
    if(props.value  !== '') {
              return(  <input className={props.className}
               onKeyPress={props.onKeyPress ? props.onKeyPress : "" } 
               onChange={props.onChange ? props.onChange : ""} />)
    } else {
        return ( <input className={props.className}
               value={props.value ? props.value : '' }
               onKeyPress={props.onKeyPress ? props.onKeyPress : "" } 
               onChange={props.onChange ? props.onChange : ""} />)
    }

}

export default TextBox;