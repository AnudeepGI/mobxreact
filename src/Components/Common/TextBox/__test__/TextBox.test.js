import React from 'react';
import ReactDOM from 'react-dom';
import TextBox from '..';

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import renderer from 'react-test-renderer';


it("rednder without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TextBox />,div)
});

it("Match Snapshot - TextBox", () => {
    const tree = renderer.create(<TextBox />).toJSON();
    expect(tree).toMatchSnapshot();
});






