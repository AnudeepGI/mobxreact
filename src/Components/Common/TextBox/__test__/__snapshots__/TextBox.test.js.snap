// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot - TextBox 1`] = `<input />`;

exports[`Match Snapshot 1 1`] = `<input />`;
